worker/task.go

```diff
@@ -244,12 +244,11 @@ func processTask(q *task.Query, gid uint32) (*task.Result, error) {
 				newValue.Val = x.Nilbyte
 			}
 			values = append(values, newValue)
-			decr() // Decrement the reference count of the pl.
 		}
 
 		filtered := types.FilterGeoUids(uids, values, geoQuery)
 		for i := 0; i < len(out.UidMatrix); i++ {
-			out.UidMatrix[i] = algo.IntersectSorted([]*task.List{out.UidMatrix[i], filtered})
+			algo.IntersectWith(out.UidMatrix[i], filtered)
```

Comment:
intersectWith mutates the first list ; which is what we want.
It is faster than IntersectSorted. Also no new allocation of list happens.